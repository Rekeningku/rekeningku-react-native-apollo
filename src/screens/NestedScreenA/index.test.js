import React from 'react'
import sinon from 'sinon'
import { shallow } from 'enzyme'
import { NestedScreenA } from './index'


describe('NestedScreenA Component', () => {
  it('renders with all props', () => {
    const wrapper = shallow(<NestedScreenA />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls open screen handler', () => {
    const handleOpenScreenB = sinon.fake()
    const wrapper = shallow(
      <NestedScreenA handleOpenScreenB={handleOpenScreenB} />,
    )
    wrapper.find('ButtonComponent').simulate('press')
    expect(handleOpenScreenB.callCount).toBe(1)
  })
})
