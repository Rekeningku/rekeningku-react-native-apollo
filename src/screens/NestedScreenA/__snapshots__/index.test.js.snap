// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedScreenA Component renders with all props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NestedScreenA />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Component)>
          Screen A
        </Styled(Component)>,
        <Styled(Component)>
          This is the first nested screen. Button that allows you to go back to previous screen was set automatically.
        </Styled(Component)>,
        <Styled(Component)>
          But you can go even deeper!
        </Styled(Component)>,
        <ButtonComponent>
          Open Nested Screen B
        </ButtonComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Screen A",
        },
        "ref": null,
        "rendered": "Screen A",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is the first nested screen. Button that allows you to go back to previous screen was set automatically.",
        },
        "ref": null,
        "rendered": "This is the first nested screen. Button that allows you to go back to previous screen was set automatically.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "But you can go even deeper!",
        },
        "ref": null,
        "rendered": "But you can go even deeper!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Open Nested Screen B",
          "onPress": undefined,
        },
        "ref": null,
        "rendered": "Open Nested Screen B",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Component)>
            Screen A
          </Styled(Component)>,
          <Styled(Component)>
            This is the first nested screen. Button that allows you to go back to previous screen was set automatically.
          </Styled(Component)>,
          <Styled(Component)>
            But you can go even deeper!
          </Styled(Component)>,
          <ButtonComponent>
            Open Nested Screen B
          </ButtonComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Screen A",
          },
          "ref": null,
          "rendered": "Screen A",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is the first nested screen. Button that allows you to go back to previous screen was set automatically.",
          },
          "ref": null,
          "rendered": "This is the first nested screen. Button that allows you to go back to previous screen was set automatically.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "But you can go even deeper!",
          },
          "ref": null,
          "rendered": "But you can go even deeper!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Open Nested Screen B",
            "onPress": undefined,
          },
          "ref": null,
          "rendered": "Open Nested Screen B",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
